### PR TITLE
Media retrieval

### PR DESCRIPTION
**Pull request by dmaroso**
Implemented http request methods with piping

### CODE REVIEW

File: chronicle-front/src/app/components/searchbar/searchbar.component.html [hunk 1]
```diff
@@ -0,0 +1,27 @@
+<mat-form-field class="example-chip-list">
+    <mat-label>Search tags</mat-label>
+    <mat-chip-list #chipList aria-label="Fruit selection">
+      <mat-chip
+        *ngFor="let fruit of fruits"
```
Comment (RevatureGentry) [inline chronicle-front/src/app/components/searchbar/searchbar.component.html]: Fruits? Let's change this to something more appropriate for this application.

File: chronicle-front/src/app/components/viewvideopage/viewvideopage.component.ts [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+  @Input() video? : Video;
+
+  constructor(private transfer : VideoPageTransferService, private mediaService : MediaRetrievalService) { }
+
+  ngOnInit(): void {
+    if (this.transfer.video) {
+      this.video = this.transfer.video;
+      this.transfer.video = undefined;
+    } else {
+     this.mediaService.getVideoById(1).subscribe(resp => {
```
Comment (RevatureGentry) [inline chronicle-front/src/app/components/viewvideopage/viewvideopage.component.ts:23]: Did we mean to hard code getting `getVideoById(1)`?

File: chronicle-front/src/app/services/media-retrieval.service.ts [hunk 3]
```diff
@@ -1,41 +1,174 @@
...
     description : 'Note',
     userId : '',
     url : 'url',
     tags : []};
-  const notes: Note[] = [n, n, n];
-  return from([notes]);
+
+
+  public getAllTags() : Observable<Tag[]>{
+    return this.httpClient.get('http://localhost:8080/myapp/videos/available-tags')
```
Comment (RevatureGentry) [inline chronicle-front/src/app/services/media-retrieval.service.ts]: Let's leverage the `environment` here instead of hard coding.

**Review by RevatureGentry — CHANGES_REQUESTED** [review]
Please change all variable names referring to fruit in the searchbar component. Let's also remove any hard-coded URL's in Services and prefer to use the Angular `environment`.

File: chronicle-front/src/app/components/searchbar/searchbar.component.html (hunk 1 above)
Comment (dmaroso) [reply chronicle-front/src/app/components/searchbar/searchbar.component.html]: We updated

File: chronicle-front/src/app/components/viewvideopage/viewvideopage.component.ts (hunk 2 above)
Comment (dmaroso) [reply chronicle-front/src/app/components/viewvideopage/viewvideopage.component.ts:23]: Just a test for functionality, will change in the future

File: chronicle-front/src/app/services/media-retrieval.service.ts (hunk 3 above)
Comment (dmaroso) [reply chronicle-front/src/app/services/media-retrieval.service.ts]: added environment import and stored urls there for consolidated access

**Review by RevatureGentry — APPROVED** [review]
💯

**Pull request merged**